apps/www/src/content/docs/components/data-table.md

```diff
@@ -20,4 +20,1137 @@ We'll start with the basic `<Table />` component and build a complex data table
+                payment,
+            }))
+        },
+    },
+]
+
+```
+
+You can access the row data using `row.original` in the `cell` function. Use this to handle actions for your row eg. use the `id` to make a DELETE call to your API.
+
+</Steps>
+
+## Pagination
+
+Next, we'll add pagination to our table.
+
+<Steps>
+
+### Update `<DataTable>`
+
+```ts:line-numbers showLineNumbers{4,12}
+import {
+    FlexRender,
+    getCoreRowModel,
+    getPaginationRowModel,
+    useVueTable,
+} from "@tanstack/vue-table"
+
+const table = useVueTable({
+    data: props.data,
+    columns: props.columns,
+    getCoreRowModel: getCoreRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+})
+```
+
+This will automatically paginate your rows into pages of 10. See the [pagination docs](https://tanstack.com/table/v8/docs/api/features/pagination) for more information on customizing page size and implementing manual pagination.
+
+### Add pagination controls
+
+We can add pagination controls to our table using the `<Button />` component and the `table.previousPage()`, `table.nextPage()` API methods.
+
+```ts:line-numbers showLineNumbers{3,15,21-39}
+// components/payments/DataTable.vue
+<script lang="ts" generic="TData, TValue">
+import { Button } from "@/components/ui/button"
+
+const table = useVueTable({
+    data: props.data,
+    columns: props.columns,
+    getCoreRowModel: getCoreRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+})
+
+</script>
+
+<template>
+    <div>
+      <div class="border rounded-md">
+        <Table>
+          { // .... }
+        </Table>
+      </div>
+      <div class="flex items-center justify-end py-4 space-x-2">
+      <Button
+          variant="outline"
+          size="sm"
+          :disabled="!table.getCanPreviousPage()"
+          @click="table.previousPage()"
+        >
+          Previous
+        </Button>
+        <Button
+          variant="outline"
+          size="sm"
+          :disabled="!table.getCanNextPage()"
+          @click="table.nextPage()"
+        >
+          Next
+        </Button>
+      </div>
+    </div>
+</template>
+
+```
+
+See [Reusable Components](#reusable-components) section for a more advanced pagination component.
+
+</Steps>
+
+## Sorting
+
+Let's make the email column sortable.
+
+<Steps>
+
+### Add the following into your `utils` file:
+
+```ts:line-numbers showLineNumbers{5,6,12-17}
+import { type ClassValue, clsx } from 'clsx'
+import { twMerge } from 'tailwind-merge'
+import { camelize, getCurrentInstance, toHandlerKey } from 'vue'
+
+import type { Updater } from '@tanstack/vue-table'
+import { type Ref } from 'vue'
+
+export function cn(...inputs: ClassValue[]) {
+  return twMerge(clsx(inputs))
+}
+
+export function valueUpdater<T extends Updater<any>>(updaterOrValue: T, ref: Ref) {
+  ref.value
+    = typeof updaterOrValue === 'function'
+      ? updaterOrValue(ref.value)
+      : updaterOrValue
+}
```

Comment:
Adding some docs for this utility is much appreciated 

User: Why do I need this utility? why I can't just use `ref`